lk2nd/smb1360/smb1360-qcom.c

```diff
@@ -15,21 +21,97 @@ static const struct smb1360_battery batteryB = {
 	.profile = SMB1360_BATTERY_PROFILE_B,
 };
 
+static int64_t abs64(int64_t val)
+{
+	return val;
+}
+
+static int smb1360_check_batt_profile(const void *fdt, int offset, uint32_t connected_rid)
+{
+	const uint32_t *prop;
+	status_t ret;
+
+	int i, len, timeout, loaded_profile, new_profile;
+	uint8_t val = 0;
+	uint32_t profile_rid_a, profile_rid_b, rid10;
+
+	prop = fdt_getprop(fdt, offset, "qcom,profile-a-rid-kohm", &len);
+	if (len < 0)
+		return -EINVAL;
+	profile_rid_a = fdt32_to_cpu(*prop);
+
+	prop = fdt_getprop(fdt, offset, "qcom,profile-b-rid-kohm", &len);
+	if (len < 0)
+		return -EINVAL;
+	profile_rid_b = fdt32_to_cpu(*prop);
+
+	rid10 = connected_rid / 10;
+
+	if ((profile_rid_a - connected_rid) < rid10) {
+		new_profile = SMB1360_BATTERY_PROFILE_A;
+	} else if ((profile_rid_b - connected_rid) < rid10) {
+		new_profile = SMB1360_BATTERY_PROFILE_B;
+	} else {
+		return -EINVAL;
+	}
+
+	return new_profile;
+}
+
+static uint32_t smb1360_parse_batt_id(const void *fdt, int offset)
+{
+	const uint32_t *prop;
+	int64_t denom, batt_id_uv;
+	int32_t len, rpull = 0, vref = 0;
+	uint32_t connected_rid;
+
+	prop = fdt_getprop(fdt, offset, "qcom,batt-id-vref-uv", &len);
+	if (len != sizeof(vref))
+		return -EINVAL;
```

Comment:
I think we can just take `return 0` as invalid value here and remove the define.